sqldelight-compiler/src/main/kotlin/app/cash/sqldelight/core/compiler/QueryGenerator.kt

```diff
@@ -170,36 +171,39 @@ abstract class QueryGenerator(
         precedingArrays.add(type.name)
         argumentCounts.add("${type.name}.size")
       } else {
-        nonArrayBindArgsCount += 1
-
-        if (!treatNullAsUnknownForEquality && type.javaType.isNullable) {
-          val parent = bindArg?.parent
-          if (parent is SqlBinaryEqualityExpr) {
-            needsFreshStatement = true
-
-            var symbol = parent.childOfType(SqlTypes.EQ) ?: parent.childOfType(SqlTypes.EQ2)
-            val nullableEquality: String
-            if (symbol != null) {
-              nullableEquality = "${symbol.leftWhitspace()}IS${symbol.rightWhitespace()}"
-            } else {
-              symbol = parent.childOfType(SqlTypes.NEQ) ?: parent.childOfType(SqlTypes.NEQ2)!!
-              nullableEquality = "${symbol.leftWhitspace()}IS NOT${symbol.rightWhitespace()}"
+        val bindParameter = bindArg?.bindParameter as? BindParameterMixin
+        if (bindParameter == null || !bindParameter.isDefault) {
```

Comment:
would this work if we just hardcoded `bindParameter.text != "DEFAULT"`? I think DEFAULT as a bind parameter is just ANSI SQL so I'm okay hardcoding that in the compiler. Then we wouldn't need the public API changes at all, we could just have the logic live here

Reply:
Sounds okay, I will do it.